File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/service/receivers/SyncStartedReceiver.java
package com.lasthopesoftware.bluewater.client.stored.service.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.lasthopesoftware.bluewater.client.stored.service.notifications.PostSyncNotification;
import com.lasthopesoftware.bluewater.client.stored.sync.StoredFileSynchronization;

import java.util.Collection;
import java.util.Collections;

public class SyncStartedReceiver extends BroadcastReceiver {

	private final PostSyncNotification syncNotification;

	public SyncStartedReceiver(PostSyncNotification syncNotification) {
		this.syncNotification = syncNotification;
	}

	@Override
	public void onReceive(Context context, Intent intent) {
		syncNotification.notify(null);
	}

	public Collection<String> acceptedEvents() {
		return Collections.singleton(StoredFileSynchronization.onSyncStartEvent);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/sync/receivers/SyncStartedReceiver.kt
package com.lasthopesoftware.bluewater.client.stored.sync.receivers

import com.lasthopesoftware.bluewater.client.stored.sync.SyncStateMessage
import com.lasthopesoftware.bluewater.client.stored.sync.notifications.PostSyncNotification

class SyncStartedReceiver(private val syncNotification: PostSyncNotification) : (SyncStateMessage.SyncStarted) -> Unit {
	override fun invoke(p1: SyncStateMessage.SyncStarted) {
		syncNotification.notify(null)
	}
}
